src/components/Home.js

```diff
@@ -1,16 +1,34 @@
 import React from 'react'
 import firebase from '../firebase_setup/firebase'
-
+import 'firebase/compat/auth'
+import 'firebase/compat/firestore'
+import 'firebase/compat/database'
 class Home extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      message: "You're logged in",
+      button: 'Logout',
+    }
+  }
+
+  //logout function
   logout() {
-    firebase.auth.signOut()
+    firebase.auth().signOut()
```

Comment:
this function call returns a promise. the network request could hypothetically fail, so `logout()` should be an async function, and you should `await firebase.auth().signOut()` before updating the state